lib/impress.client.js

```diff
@@ -1007,25 +1021,25 @@ Client.prototype.signOut = function(
   const client = this;
   const application = client.application;
 
-  if (client.session) {
-    const login = client.session.login;
-    if (login) {
-      const user = application.users.get(login);
-      if (user && user.sessions) {
-        delete user.sessions[client.sid];
-      }
-    }
-    if (client.session.login) {
-      delete client.session.login;
-    }
-    if (client.session.group) {
-      delete client.session.group;
+  if (!client.session) return callback(false);
+
+  const login = client.session.login;
+  if (login) {
+    const user = application.users.get(login);
+    if (user && user.sessions) {
+      delete user.sessions[client.sid];
     }
-    client.sessionModified = true;
-    client.logged = false;
-    client.user = null;
-    callback(true);
-  } else callback(false);
+  }
+  if (client.session.login) {
+    delete client.session.login;
```

Comment:
It's better to set it to `null` so that it doesn't mess with V8's cached object forms and doesn't cause deopts.